internal/provider/be_api_key_resource.go

```diff
@@ -155,6 +156,14 @@ func (r *beApiKeyResource) Read(ctx context.Context, req resource.ReadRequest, r
 	// retrieve the be api key from PropelAuth
 	beApiKeyInfo, err := r.client.GetBeApiKeyInfo(state.Environment.ValueString(), state.ApiKeyId.ValueString())
 	if err != nil {
+		// If error is not found, it indicates that the resource should be deleted.
+		// It is expected that the error response will be in this exact string when not found.
+		if strings.Contains(err.Error(), "{\"error_code\":\"not_found\",\"user_facing_error\":\"Not found.\"}") {
```

Comment:
@itailevi98 Kind of a nit but worth changing: Let's just check the value of the `error_code`, as this shouldn't ever change whereas the error message should be able to change in the future as needed.

Maybe we could even have an `ErrorMessage` struct we could try marshaling error response bodies into and then converting that to a more TF-specific error within `GetBeApiKeyInfo`...?